thetis/solver.py

```diff
@@ -1077,20 +1078,43 @@ def load_state(self, i_export, outputdir=None, t=None, iteration=None):
         for e in self.exporters.values():
             e.set_next_export_ix(self.i_export + offset)
 
-    def print_state(self, cputime):
+    def print_state(self, cputime, print_header=False):
         """
         Print a summary of the model state on stdout
 
-        :arg float cputime: Measured CPU time
+        :arg float cputime: Measured CPU time in seconds
+        :kwarg print_header: Whether to print column header first
         """
-        norm_h = norm(self.fields.elev_2d)
-        norm_u = norm(self.fields.uv_3d)
-
-        line = ('{iexp:5d} {i:5d} T={t:10.2f} '
-                'eta norm: {e:10.4f} u norm: {u:10.4f} {cpu:5.2f}')
-        print_output(line.format(iexp=self.i_export, i=self.iteration,
-                                 t=self.simulation_time, e=norm_h,
-                                 u=norm_u, cpu=cputime))
+        # generate simulation time string
+        if self.options.simulation_initial_date is not None:
+            now = self.options.simulation_initial_date + datetime.timedelta(seconds=self.simulation_time)
+            time_str = f'{now:%Y-%m-%dT%H:%M:%S%Z}'
+        else:
+            time_str = f'{self.simulation_time:.2f}'.rjust(22)
+
+        norm_h = norm(self.fields.solution_2d.split()[1])
+        norm_u = norm(self.fields.solution_2d.split()[0])
+
+        # list of entries to print: (header, value, format)
+        entries = [
+            ('exp', self.i_export, '5d'),
+            ('iter', self.iteration, '5d'),
+            ('time', time_str, '22s'),
+            ('eta norm', norm_h, '12.4f'),
+            ('u norm', norm_u, '12.4f'),
+            ('Tcpu', cputime, '6.2f'),
+        ]
+
+        if print_header:
+            # generate header
+            widths = [len(('{:' + e[2] + '}').format(e[1])) for e in entries]
+            header = ' '.join([e[0].rjust(w) for e, w in zip(entries, widths)])
```

Comment:
You can get this with
`header = ' '.join([e[0].rjust(len(f'{e[1]:{e[2]}}')) for e in entries])`